android/hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/MainActivity.java

```diff
@@ -1,27 +1,68 @@
 package com.sdl.hellosdlandroid;
 
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
 import android.view.Menu;
 import android.view.MenuItem;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import static android.Manifest.permission.BLUETOOTH_CONNECT;
 
 public class MainActivity extends AppCompatActivity {
 
+    private static final int REQUEST_CODE = 200;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        //If we are connected to a module we want to start our SdlService
+
+
         if (BuildConfig.TRANSPORT.equals("MULTI") || BuildConfig.TRANSPORT.equals("MULTI_HB")) {
-            SdlReceiver.queryForConnectedService(this);
-        } else if (BuildConfig.TRANSPORT.equals("TCP")) {
+            if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                if (!checkPermission()) {
+                    requestPermission();
+                }
+            } else if (BuildConfig.TRANSPORT.equals("TCP")){
+                //If we are connected to a module we want to start our SdlService
+                SdlReceiver.queryForConnectedService(this);
+            }
+        } else {
```

Comment:
```suggestion
            if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
                    && !checkPermission()) {
                requestPermission();
                return;
            }
            //If we are connected to a module we want to start our SdlService
            SdlReceiver.queryForConnectedService(this);

        } else if (BuildConfig.TRANSPORT.equals("TCP")) {
```

Reply:
This logic needed to be cleaned up. Basically check for OS version and if permission is granted, if not request them. If OS is less than 12 or permissions are granted, query for the RS. The previous else if was in the wrong position for TCP.